{{cookiecutter.project_name}}/pyproject.toml

```diff
@@ -108,36 +119,40 @@ port.exclude_lines = [
 ]
 
 [tool.ruff]
-select = [
-  "E", "F", "W", # flake8
-  "B",           # flake8-bugbear
-  "I",           # isort
-  "ARG",         # flake8-unused-arguments
-  "C4",          # flake8-comprehensions
-  "EM",          # flake8-errmsg
-  "ICN",         # flake8-import-conventions
-  "ISC",         # flake8-implicit-str-concat
-  "G",           # flake8-logging-format
-  "PGH",         # pygrep-hooks
-  "PIE",         # flake8-pie
-  "PL",          # pylint
-  "PT",          # flake8-pytest-style
-  "PTH",         # flake8-use-pathlib
-  "RET",         # flake8-return
-  "RUF",         # Ruff-specific
-  "SIM",         # flake8-simplify
-  "T20",         # flake8-print
-  "UP",          # pyupgrade
-  "YTT",         # flake8-2020
-  "EXE",         # flake8-executable
-  "NPY",         # NumPy specific rules
-  "PD",          # pandas-vet
-]
```

Comment:
Question: should we provide these rules but exclude them by commenting them out?

Reply:
I think we should keep all of them un-commented. Is any of these rules causing any trouble in the CI?

Reply:
Some of them were causing trouble because the code is copied from PyBaMM, which does not adhere to these rules. Removing them from the project's `pyproject.toml` is fine, personally. We should provide them in the template so that people can un-comment them at their will if needed.

Reply:
Oh yes, my bad. I meant the same thing. Keep in the template and remove as per our requirements from the project.

Reply:
The reason I say this is because let us say in the future the entry points are updated upstream in the `PyBaMM` repository and we would want to update that in the template as well but the rules might alter the way it is written so I feel ignoring them would be a better option because most of them would already be linted upstream, rather we could add those files first in `pybamm_cookiecutter` project as it has the same linting rules as `PyBaMM` and then move it inside the template.

Reply:
I see no harm in doing this; should be okay imo.